### PR TITLE
UI Skeleton

### PR DESCRIPTION
**Pull request by Akshit-Accolite**
added the UI skeleton

### CODE REVIEW

File: src/main/java/com/erstudio/ERSUI.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+import javax.swing.filechooser.FileSystemView;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class ERSUI {
+
+    JLabel labelTSString, labelVersionString, labelNewVersionString, labelGUID;
+    JTextField textFieldTs, textVersionString, textNewVersionString, textGUID;
+    JButton buttonTS, buttonGenerate, jButton, jButton1;
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Naming.

File: src/main/java/com/erstudio/ERSUI.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    ButtonGroup buttonGroup;
+    JRadioButton jRadioButton, jRadioButton1;
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Naming

File: src/main/java/com/erstudio/ERSUI.java [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    JTextArea textArea;
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Naming.

File: src/main/java/com/erstudio/ERSUI.java [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    JFrame frame;
+    JTabbedPane tp;
+    JPanel panel;
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Naming

File: src/main/java/com/erstudio/ERSUI.java [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+    public JPanel initTeamServerUI() {
+        panel = new JPanel();
+        panel.setLayout(new GridBagLayout());
+
+        labelTSString = new JLabel("Please choose TS main folder");
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: All the string constants needs to be in separate file with final static.

File: src/main/java/com/erstudio/ERSUI.java [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+        c.gridwidth = 1;
+        c.weightx = .5;
+        c.weighty = .2;
+        c.gridx = 2;
+        c.gridy = 1;
+        c.ipadx = 50;
+        c.ipady = 0;
+        panel.add(buttonTS, c);
+
+        labelVersionString = new JLabel("Version String to be Replaced");
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Add String constant.

File: src/main/java/com/erstudio/ERSUI.java [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+        c.fill = GridBagConstraints.HORIZONTAL;
+        c.ipady = 100;      //make this component tall
+        c.weightx = 0.0;
+        c.gridwidth = 3;
+        c.gridx = 0;
+        c.gridy = 7;
+        panel.add(textArea, c);
+
+        textArea.setBorder(BorderFactory.createLineBorder(Color.BLACK));
+
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Extra line

File: src/main/java/com/erstudio/ERSUI.java [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+        return panel;
+    }
+
+    public JPanel initDAUI() {
+        JPanel panel = new JPanel();
+
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Extra line.

File: src/main/java/com/erstudio/ERSUI.java [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+        return panel;
+    }
+
+    public JPanel inithelpUI() {
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Naming

File: src/main/java/com/erstudio/ERSUI.java [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+        c.gridy = 7;
+        panel.add(textArea, c);
+
+        textArea.setBorder(BorderFactory.createLineBorder(Color.BLACK));
+
+
+        return panel;
+    }
+
+    public JPanel initDAUI() {
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Naming.

File: src/main/java/com/erstudio/ERSUI.java [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+        int r = j.showOpenDialog(null);
+
+        if (r == JFileChooser.APPROVE_OPTION) {
+            // set the label to the path of the selected directory
+            textFieldTs.setText(j.getSelectedFile().getAbsolutePath());
+        }
+    }
+
+
+    ERSUI() {
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSUI.java]: Why Default modifier??

File: src/main/java/com/erstudio/ERStudioReleaseHelper.java [hunk 12]
```diff
@@ -13,9 +13,11 @@
 	 * Launches ERStudio Helper Application
 	 * @param args
 	 */
+
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERStudioReleaseHelper.java:16]: Why Extra line.

**Review by vaibhavjain3 — CHANGES_REQUESTED** [review]
Review.

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java [hunk 13]
```diff
@@ -0,0 +1,243 @@
+package main.java.com.erstudio;
+
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java:2]: Extra line.

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java [hunk 14]
```diff
@@ -0,0 +1,243 @@
...
+
+/**
+ * @author akshit.arora
+ * for the UI Skeleton
+ */
+
+import main.java.com.erstudio.constants.Constants;
+
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: Extra line.

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java [hunk 15]
```diff
@@ -0,0 +1,243 @@
...
+    JLabel labelTSString, labelVersionString, labelNewVersionString, labelGUID;
+    JTextField textFieldTs, textVersionString, textNewVersionString, textGUID;
+    JButton buttonBrowse, buttonGenerate, buttonChangeVersion, buttonReset;
+    ButtonGroup buttonGroupUpgradeType;
+    JRadioButton jRadioButtonFullUpgrade, jRadioButtonPatchUpgrade;
+    JTextArea textAreaStatus;
+    JFrame mainFrame;
+    JTabbedPane tabbedPaneUI;
+    JPanel panelTeamServer, panelDataHelper, panelHelp;
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: panelDataArchitect

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java [hunk 16]
```diff
@@ -0,0 +1,243 @@
...
+
+        // invoke the showsOpenDialog function to show the save dialog
+        int r = j.showOpenDialog(null);
+
+        if (r == JFileChooser.APPROVE_OPTION) {
+            // set the label to the path of the selected directory
+            textFieldTs.setText(j.getSelectedFile().getAbsolutePath());
+        }
+    }
+
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java:215]: Extra line.

**Review by vaibhavjain3 — CHANGES_REQUESTED** [review]

File: src/main/java/com/erstudio/ERSUI.java (hunk 1 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 2 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 3 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 4 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 5 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 8 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 9 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 11 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java (hunk 13 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java:2]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 10 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERStudioReleaseHelper.java (hunk 12 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERStudioReleaseHelper.java:16]: done

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java (hunk 14 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: done

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java (hunk 15 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: done

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java (hunk 16 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java:215]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 6 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSUI.java (hunk 7 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSUI.java]: done

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class ERSStudioReleaseHelperUI {
+
+    JLabel labelTSString, labelVersionString, labelNewVersionString, labelGUID;
+    JTextField textFieldTs, textVersionString, textNewVersionString, textGUID;
+    JButton buttonBrowse, buttonGenerate, buttonChangeVersion, buttonReset;
+    ButtonGroup buttonGroupUpgradeType;
+    JRadioButton jRadioButtonFullUpgrade, jRadioButtonPatchUpgrade;
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: use radioButtonFullUpgrade. like you are using buttonBrowse, textFieldTs.

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java [hunk 18]
```diff
@@ -0,0 +1,239 @@
...
+        panelDataArchitect = new JPanel();
+        return panelDataArchitect;
+    }
+
+    public JPanel initHelpUI() {
+        panelHelp = new JPanel();
+        return panelHelp;
+    }
+
+    //on click the of browse button
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: lets remove this comment,  onPressedBrowse specifies the thing.

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java [hunk 19]
```diff
@@ -0,0 +1,239 @@
...
+    public void onPressedBrowse() {
+        JFileChooser j = new JFileChooser(FileSystemView.getFileSystemView().getHomeDirectory());
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: very bad variable name.

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java [hunk 20]
```diff
@@ -0,0 +1,239 @@
...
+
+        // set the selection mode to directories only
+        j.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+
+        // invoke the showsOpenDialog function to show the save dialog
+        int r = j.showOpenDialog(null);
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: bad naming of variable. Based on what showOpenDialog returns. rename r.

File: src/main/java/com/erstudio/constants/Constants.java [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+    public static final String STRING_TO_BE_REPLACED = "Version String to be Replaced";
+    public static final String NEW_VERSION_STRING = "New Version String";
+    public static final String GUID = "GUID";
+    public static final String GENERATE = "Generate";
+    public static final String FULL_UPGRADE = "Full Upgrade";
+    public static final String PATCH_UPGRADE = "Patch Upgrade";
+    public static final String CHANGE_VERSION_NUMBER = "Change Version Number";
+    public static final String RESET = "Reset";
+    public static final String TEAM_SERVER = "TeamServer";
+    public static final String DATA_HELPER = "Data Helper";
```
Comment (vaibhavjain3) [inline src/main/java/com/erstudio/constants/Constants.java]: It's Data Architect.

**Review by vaibhavjain3 — CHANGES_REQUESTED** [review]

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java (hunk 19 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: done

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java (hunk 18 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: done

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java (hunk 17 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: done

File: src/main/java/com/erstudio/constants/Constants.java (hunk 21 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/constants/Constants.java]: done

File: src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java (hunk 20 above)
Comment (Akshit-Accolite) [reply src/main/java/com/erstudio/ERSStudioReleaseHelperUI.java]: done

**Pull request merged**